### PR TITLE
Ria 3163 payment after submit appeal

### PR DESCRIPTION
**Pull request by bradsorour**
### JIRA link (if applicable) ###

[RIA-3163](https://tools.hmcts.net/jira/browse/RIA-3163)

### Change description ###

See Jira

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/domain/handlers/presubmit/PaymentAppealHandler.java [hunk 1]
```diff
@@ -189,8 +166,79 @@ private void writeFeeDetailsToCaseData(AsylumCase asylumCase, Fee fee) {
...
 
-    private PaymentResponse makePayment(CreditAccountPayment creditAccountPayment) {
+    public PaymentResponse makePayment(CreditAccountPayment creditAccountPayment) {
+
+        requireNonNull(creditAccountPayment, "creditAccountPayment must not be null");
+
+        try {
+            return paymentService.creditAccountPayment(creditAccountPayment);
+        } catch (FeignException fe) {
+            log.error("Payment failed: {}", fe.getMessage());
```
[the remark this answers is not in the document]
Comment (bradsorour) [reply src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/domain/handlers/presubmit/PaymentAppealHandler.java:176]: These are automated GitHub bot messages from RIA-3082. Not planning to fix these right now as the current unit test pass Sonar code coverage. Fixing these new bot warnings interrogating the try/catch blocks are verifying the error messages written to the logs. The new GitHub bot warnings were covered off with Marcin in RIA-3082.

File: src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/domain/handlers/presubmit/PaymentAppealHandler.java [hunk 2]
```diff
@@ -189,8 +166,79 @@ private void writeFeeDetailsToCaseData(AsylumCase asylumCase, Fee fee) {
...
+            try {
+                return objectMapper.readValue(fe.contentUTF8(), PaymentResponse.class);
+            } catch (JsonProcessingException je) {
+                log.error("Error parsing the failed payment response: {}", je.getMessage());
+            }
+        }
+        return null;
```
[the remark this answers is not in the document]
Comment (bradsorour) [reply src/main/java/uk/gov/hmcts/reform/iacasepaymentsapi/domain/handlers/presubmit/PaymentAppealHandler.java:183]: Same as above.

**Review by Srihari-Nakkala — APPROVED** [review]
LGTM

**Review by makowalski — APPROVED** [review]
LGTM

**Pull request merged**